editor/io_plugins/editor_atlas.cpp

```diff
@@ -76,68 +77,90 @@ void EditorAtlas::fit(const Vector<Size2i> &p_rects, Vector<Point2i> &r_result,
 		if (w < widest)
 			continue;
 
-		Vector<int> hmax;
-		hmax.resize(w);
-		for (int j = 0; j < w; j++)
-			hmax[j] = 0;
-
-		//place them
-		int ofs = 0;
+        Vector<int> wmax;
+        wmax.resize(Stotal/w);
+        for (int j = 0; j < wmax.size(); j++)
+            wmax[j] = 0;
 
 		for (int j = 0; j < wrects.size(); j++) {
 
-			if (ofs + wrects[j].s.width > w) {
-
-				ofs = 0;
-			}
-
-			int from_y = 0;
-			for (int k = 0; k < wrects[j].s.width; k++) {
-
-				if (hmax[ofs + k] > from_y)
-					from_y = hmax[ofs + k];
-			}
-
-			wrects[j].p.x = ofs;
-			wrects[j].p.y = from_y;
-
-			int end_h = from_y + wrects[j].s.height;
-			int end_w = ofs + wrects[j].s.width;
-
-			for (int k = 0; k < wrects[j].s.width; k++) {
-
-				hmax[ofs + k] = end_h;
-			}
+            int newX = 0;
+			int newY = 0;
+
+            int pieceW = wrects[j].s.width;
+            int pieceH = wrects[j].s.height;
+
+            bool foundPlace;
+
+            do {
+                foundPlace = true;
+                newX = 0;
+                if(wmax.size() <= newY + pieceH) {
```

Comment:
There are some style issues, e.g. missing space between `if` and `(`. Check docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html for more info on the code style and how to apply clang-format locally.